rewindable-index/src/RewindableIndex/Storable.hs

```diff
@@ -231,10 +250,42 @@ getEvents
   -> StorableMonad h [StorableEvent h]
 getEvents s = do
   memoryEs <- getMemoryEvents (s ^. storage)
-              & V.freeze <&> V.toList
+              & V.freeze <&> foldEvents . V.toList
   diskEs   <- getStoredEvents (s ^. handle)
   pure $ diskEs ++ memoryEs
 
+{- This function is used to add a checkpoint to the in-memory part of event indexers.
+   You can use checkpoints to add synthetic events to the memory buffer. Filling the
+   memory buffer will flush the real events to disk faster than not using checkpoints.
+
+   This is useful if the number of events is small and you run the risk of never
+   storing them on-disk. When that happens, if the indexer is restarted it will
+   resume from the Genesis block, which is something that is not an expected
+   behaviour.
+
+   Possible future uses of checkpoints:
+
+   * We currently assume that the indexer has processed blocks up to the latest event
+     processed. In case there are very few events this assumption introduces a big
+     error. We may eventually use checkpoints to signal to the indexer that the latest
+     block is more recent than the latest event processed.
+
+   * Currently synthetic events cannot be stored in the persistent database. We may
+     decide to offer this option to implementers if there is interest. This would
+     allow for faster resumes if the last stored event is a lot older than the
+     currently processed block (similar to the in-memory description of checkpoints).
+-}
+checkpoint
```

Comment:
Looking at this function, seems like it will be up to the user to call this at different points in time while running the indexer right?

Reply:
Yes. I've had a discussion with @kayvank where I had the idea of putting the event filtering function within the indexer, in which case this function could be called automatically. However, we can make that change later if we choose to do so, so I did not include it in this PR.

Reply:
I think that will be a nice changes. Doing the filtering as its done with the Utxos currently, is adding a lot of noise to the Indexers.hs module. +1 for me when this change comes in